### DISCUSSION TITLE
making a system be called before bevy draws stuff

### DISCUSSION THREAD

**Discussion by BipedPotato** [bevyengine/bevy#5256] [Q&A]
So basically i am making a collision system and it is contained in a ecs system. Is there a way i can make this system be called before bevy draws stuff because there is a lot of stuttering because bevy seems to be drawing stuff before this system is called.

**Comment by Lum3s** [comment]
Is something like this what you are looking for?

[Unofficial Bevy Cheat Book - Stages](https://bevy-cheatbook.github.io/programming/stages.html#stages)

Alternatively if you want stageless, there is a plugin for that [https://github.com/IyesGames/iyes_loopless](https://github.com/IyesGames/iyes_loopless) and an [open RFC](https://github.com/bevyengine/rfcs/pull/45) which should be due soon.

**Comment by meowitzher** [comment]
I had the problem of stuttering in my collision system when there was ambiguity with running my collision system and bevy's `propagate_transform` system in the `PostUpdate` stage. I am currently ensuring that everything runs before this system, and I am not having any issues with stuttering or jitter currently.

Here's how I am doing this for reference:
```rust 
        app.add_system_to_stage
        (          
            CoreStage::PostUpdate, 
            collision_system
            .before(bevy::transform::transform_propagate_system)
        );
```
Hopefully that helps! 

I solved this with the help from other kind users in [this thread](https://github.com/bevyengine/bevy/discussions/5067), which should help you to further debug the problems you are having if the above doesn't work!
